src/InputParser.jl

```diff
@@ -0,0 +1,374 @@
+module InputParser
+
+# Julia implicitly numbers Enum items 0,1,2,3,4 (in order)
+# I put them in same order as FORTRAN code (i.e. NOPT=0 meant
+# ConsolidationSwell so Int(ConsolidationSwell) = 0) however
+# this wont matter since we should never be referring to the integer 
+# values explicitly anyways
+@enum Model ConsolidationSwell LeonardFrost Schmertmann CollapsibleSoil SchmertmannElastic 
+
+# These are variables/objects visible to all modules
+# that include InputParser module
+export Model, InputData
+
+# This code repeats many times in InputData constructor
+# Parses String at input[index] and returns an array A such that size(A) == items
+function parseCurrentLine(input::Array{String}, items::Int, index::Int)
+    currentLine = input[index]
+    # Tabs ruin splitting by space
+    currentLine = replace(currentLine, "\t" => " ")
+    currentLineData = split(currentLine, " ")
+    currentLineData = filter(x -> x != "", currentLineData)
+    # Check if data is there
+    if size(currentLineData)[1] < items
+        println("Error: Invalid input file!")
+        println("\t>Line $(index): $(items) values expected")
+        return -1
+    end
+    return currentLineData
+end
+
+# TODO: Search for error handling (throws Exception/ try-catch block)
+# to make InputData constructor more robust. Many Type errors are 
+# possible when converting input file
+
+# struct for InputData (OOP in Julia)
+struct InputData
+    problemName::String
+    numProblems::Int
```

Comment:
Does Julia have a natural number type?  (It probably doesn't, but sometimes when we say integer, we really mean a natural number (or whole number), since negative values don't make sense.)

Reply:
There is type `UInt`, I can change all counter variables to `UInt`

Reply:
Yes, lets use an unsigned int when the data cannot take a negative value.

Reply:
The `UInt` seems to print hex values by default

Reply:
Okay, just go back to using integers.  I did a bit of a google, and it isn't worth it to have to convert to decimal.  We won't actually display these values very often, but it will be confusing when we do.  I think uint is added to Julia for saving memory, not to provide an actual natural number type.  We might as well do what everybody else does and use integers to represent what we know are natural numbers.  :smile: